src/sentry/integrations/utils/code_mapping.py

```diff
@@ -155,45 +156,15 @@ def __init__(self, frame_file_path: str) -> None:
         self.extension = get_extension(frame_file_path)
         if not self.extension:
             raise UnsupportedFrameFilename("It needs an extension.")
-        if self.frame_type() == "packaged":
-            self._packaged_logic(frame_file_path)
-        else:
-            self._straight_path_logic(frame_file_path)
-
-    def frame_type(self) -> str:
-        type = "packaged"
-        if self.extension not in ["py"]:
-            type = "straight_path"
-        return type
-
-    def _packaged_logic(self, frame_file_path: str) -> None:
-        self.root, self.file_and_dir_path = frame_file_path.split("/", 1)
 
-        # Check that it does have at least a dir
-        if self.file_and_dir_path.find("/") > -1:
-            self.dir_path, self.file_name = self.file_and_dir_path.rsplit("/", 1)
+        start_at_index = get_straight_path_prefix_end_index(frame_file_path)
+        if start_at_index == 0:
+            self.root = frame_file_path.split("/")[0]
         else:
-            # A package name, a file but no dir (e.g. requests/models.py)
-            self.dir_path = ""
-            self.file_name = self.file_and_dir_path
```

Comment:
The only class parameter that needs to be defined is `self.root`. We do not need to branch the logic and create a whole new function in the class constructor just to define this one variable.